src/main/java/org/pentaho/di/trans/dataservice/optimization/cache/ServiceObserver.java

```diff
@@ -68,7 +68,11 @@ public ServiceObserver( DataServiceExecutor executor ) {
         if ( isRunning ) {
           latch = new CountDownLatch( 1 );
           try {
-            latch.await( 1, TimeUnit.SECONDS );
+            while ( !latch.await( 1, TimeUnit.SECONDS ) ) {
+              if ( !isRunning ) {
```

Comment:
Of course you could just `break` but I was questioning the `if` itself. It's ok.